controller/api/models.py

```diff
@@ -1129,10 +1131,12 @@ def _etcd_purge_user(**kwargs):
         pass
 
 
-def _etcd_create_app(**kwargs):
+def _etcd_publish_app(**kwargs):
     appname = kwargs['instance']
-    if kwargs['created']:
+    try:
         _etcd_client.write('/deis/services/{}'.format(appname), None, dir=True)
+    except KeyError:
```

Comment:
I just double-checked myself.  That was exactly the error I was swallowing-- deliberately.  If the directory already exists, the "mkdir" on line 1137 fails.  As it happens, if that particular line failed for any _other_ reason that also throws a `KeyError`, we shouldn't care too much.  The same directory that line is attempting to create would actually be created implicitly by the `deis-publisher` anyway as it goes about its business of making container host IPs and ports discoverable by writing `/deis/services/<app>/<container>/<address>`.

I will add a comment in here to clarify this.